mindmeld/models/taggers/pytorch_crf.py

```diff
@@ -575,11 +608,24 @@ def train_one_epoch(self, train_dataloader):
         self.train()
         train_loss = 0
         for batch_idx, (inputs, mask, labels) in enumerate(train_dataloader):
-            self.optim.zero_grad()
-            loss = self.forward(inputs, labels, mask, drop_input=self.drop_input)
-            train_loss += loss.item()
-            loss.backward()
-            self.optim.step()
+            def closure():
+                nonlocal train_loss
+                self.optim.zero_grad()
+                # pylint: disable=cell-var-from-loop
+                loss = self.forward(inputs, labels, mask, drop_input=self.drop_input)
+                if self.reg_weight > 0:
+                    if self.optimizer == "lbfgs":
+                        loss += self.compute_regularized_loss(l1=False)
+                    loss += self.compute_regularized_loss(l1=True)
```

Comment:
It feels a little weird to hard-code that they always use balanced L1+L2. Is there any problem with using separate `reg_weight_l1` and `reg_weight` (L2)?

Reply:
Nope. You're right! Ideally we should have separate params for both! I've fixed that!